lib/core/src/server/build-dev.js

```diff
@@ -124,6 +124,78 @@ const updateCheck = async (version) => {
   return result;
 };
 
+// We only expect to have release notes available for major and minor releases.
+// For this reason, we convert the actual version of the build here so that
+// every place that relies on this data can reference the version of the
+// release notes that we expect to use.
+const getReleaseNotesVersion = (version) => {
+  const { major, minor } = semver.parse(version);
+  const { version: releaseNotesVersion } = semver.coerce(`${major}.${minor}`);
+  return releaseNotesVersion;
+};
+
+const getReleaseNotesFailedState = (version) => {
+  return {
+    success: false,
+    currentVersion: getReleaseNotesVersion(version),
+    showOnFirstLaunch: false,
+  };
+};
+
+export const RELEASE_NOTES_CACHE_KEY = 'releaseNotesData';
+
+export const getReleaseNotesData = async (currentVersionToParse, fileSystemCache) => {
```

Comment:
Of note here is that I decided to pass the cache in so it would be easier to mock when testing. I can never figure out how to mock a node module w/ jest & change the return value in each test.

Reply:
@kylesuss have you tried https://jestjs.io/docs/en/manual-mocks ?

Reply:
I didn't. Mostly a lot of combinations of `jest.mock`, `jest.doMock`, `jest.mockReturnValue`, etc.

> If the module you are mocking is a Node module (e.g.: lodash), the mock should be placed in the __mocks__ directory adjacent to node_modules (unless you configured roots to point to a folder other than the project root) and will be automatically mocked. There's no need to explicitly call jest.mock('module_name').

Do we want to mock this module all the time? That seems a bit heavy handed.

Reply:
@kylesuss something like

```js
import foo from 'bar';

jest.mock('bar');

it('...', () => {
  foo.mockReturnValue('...');
});
```

Is that what you mean?

Reply:
Yeah I tried that.

```js
import Cache from 'file-system-cache';
import { getReleaseNotesData, RELEASE_NOTES_CACHE_KEY } from './build-dev';

jest.mock('file-system-cache');

it('handles errors gracefully', async () => {
  Cache.mockReturnValue(() => ({}));
  // The rest of the test
});
```

The `cache` in the module is `undefined`.

This works:

```js
jest.mock('file-system-cache', () => ({
  __esModule: true,
  default: () => ({}),
}));
```

But it means every test will have the same return value. I can't seem to override it with `mockReturnValue` inside the test.

Jest's docs [suggest](https://jestjs.io/docs/en/jest-object#jestdomockmodulename-factory-options) using `doMock` in these cases for ES6 imports so I tried this:

```js
it('handles errors gracefully', async () => {
    const version = '4.0.0';
    jest.doMock('file-system-cache', () => ({
      __esModule: true,
      default: jest.fn(() => ({})),
    }));

    return import('file-system-cache').then(async () => {
      expect(await getReleaseNotesData(version)).toEqual({
        currentVersion: version,
        showOnFirstLaunch: false,
        success: false,
      });
    });
  });
```

But `cache` doesn't end up having the mock value -- it is just the module's normal FileSystemCache object.

So I just tried manual mocks as suggested by @lifeiscontent:

```js
// __mocks__/file-system-cache.js
module.exports = jest.fn(() => ({ a: 'b' }));

// build-dev.test.js
it('handles errors gracefully', async () => {
  Cache.mockReturnValue(() => ({}));
  // The rest of the test
});
```

And the value is `({ a: 'b' })`.

Haha. So I donno -- maybe I am missing something but I have tried quite a lot.